EKF/mag_control.cpp

```diff
@@ -72,6 +72,8 @@ void Ekf::controlMagFusion()
 		return;
 	}
 
+	_mag_yaw_reset_req |= otherHeadingSourcesHaveFinished();
```

Comment:
This returns true only at the one time point where the other sensor is lost. If in the following lines, the reset is not happening due to MagFieldDisturbance or another reason. It will still try to reset, as the `_mag_yaw_reset_req` stays true. So that is okay.